plugin/src/templates/inject/source-nodes-api-runner.js

```diff
@@ -0,0 +1,173 @@
+/**
+ * This file is used to override node sourcing in Gatsby core so we can add decoupled sourcing to Gatsby
+ * This file was copied from Gatsby core and modified and will be copied back into node_modules/gatsby when Netlify does a cold-cache build.
+ * See 'helpers/files.ts'
+ */
+
+const GatsbyCacheLmdb = require(`./cache-lmdb`).default
+const { getDataStore } = require(`../datastore`)
+const reporter = require(`gatsby-cli/lib/reporter`)
+
+const SOURCE_PLUGIN_ALLOW_LIST = [
+  `gatsby-source-wordpress`,
+  `gatsby-source-drupal`,
+  `gatsby-source-contentful`,
+]
+
+let synchronizerCache
+let originalSourceNodesApiRunner
+let isFirstSource = true
+
+exports.sourceNodesApiRunner = async function sourceNodesApiRunner(args) {
+  const usingMerlin =
+    process.env.DECOUPLED_SOURCING === `true` ||
+    process.env.DECOUPLED_SOURCING === `1`
+
+  if (!originalSourceNodesApiRunner) {
+    originalSourceNodesApiRunner = require(`${__filename}.original`)
+  }
+
+  if (!usingMerlin) {
+    return originalSourceNodesApiRunner.sourceNodesApiRunner(args)
+  }
+
+  const { synchronize } = require(`@gatsby-cloud-pkg/merlin-synchronizer`)
```

Comment:
We _may_ need to replace this with a full resolved path to the dependency if resolving from an auto-installed plugin fails